urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/io/fileio/FileCopy.java

```diff
@@ -0,0 +1,254 @@
+package ikrs.io.fileio;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * The standard java libraries have no file- and directory- copy routines.
+ *
+ * This class is a simple solution.
+ *
+ * @author Ikaros Kappler
+ * @date 2012-12-19
+ * @version 1.0.0
+ **/
+
+
+public class FileCopy {
+
+    /**
+     * The default buffer size to use (in bytes).
+     **/
+    public static final int DEFAULT_BUFFER_SIZE = 1024;
+
+    
+    /**
+     * Copy the data from source to destination.
+     * Source and destination MUST be regular files or directories.
+     *
+     * There are four possible cases. Example:
+     *  (i)   source      is a regular file,
+     *        destination is a regular file.
+     *       
+     *         Then the bytes from source will be written into the destination.
+     *         If the destination already exists it will be overwritten.
+     *
+     *  (ii)  source is a directory,
+     *        destination is a regular file.
+     *
+     *         This will cause an IllegalArgumentException to be thrown!
+     *  
+     *  (iii) source is a regular file,
+     *        destination is a directory.
+     *
+     *         A new file will be created inside the destination directory having the
+     *         same name as source.
+     *
+     *  (iv)  source is a directory,
+     *        destination is a directory.
+     *
+     *         The destination directory will be created (if not exists) and all files
+     *         and sub directories from source will be copied into the destination 
+     *         directory recursively.
+     *
+     **/
+    public static boolean copy( File source,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.